src/Projection/Projection/Store/InMemoryStore.php

```diff
@@ -25,22 +37,68 @@ public function get(string $projectionId): Projection
         throw new ProjectionNotFound($projectionId);
     }
 
-    public function all(): ProjectionCollection
+    /** @return list<Projection> */
+    public function find(ProjectionCriteria|null $criteria = null): array
     {
-        return new ProjectionCollection(array_values($this->projections));
+        $projections = array_values($this->projections);
+
+        if ($criteria === null) {
+            return $projections;
+        }
+
+        return array_values(
+            array_filter(
+                $projections,
+                static function (Projection $projection) use ($criteria): bool {
+                    if ($criteria->ids !== null) {
+                        if (!in_array($projection->id(), $criteria->ids, true)) {
+                            return false;
+                        }
+                    }
+
+                    if ($criteria->groups !== null) {
+                        if (!in_array($projection->group(), $criteria->groups, true)) {
+                            return false;
+                        }
+                    }
+
+                    if ($criteria->status !== null) {
+                        if (!in_array($projection->status(), $criteria->status, true)) {
+                            return false;
+                        }
+                    }
+
+                    return true;
+                },
+            ),
+        );
     }
 
-    public function save(Projection ...$projections): void
+    public function add(Projection $projection): void
     {
-        foreach ($projections as $projection) {
-            $this->projections[$projection->id()] = $projection;
+        if (array_key_exists($projection->id(), $this->projections)) {
+            throw new ProjectionAlreadyExists($projection->id());
         }
+
+        $this->projections[$projection->id()] = $projection;
     }
 
-    public function remove(string ...$projectionIds): void
+    public function update(Projection $projection): void
     {
-        foreach ($projectionIds as $projectionId) {
-            unset($this->projections[$projectionId]);
+        if (!array_key_exists($projection->id(), $this->projections)) {
+            throw new ProjectionNotFound($projection->id());
         }
+
+        $this->projections[$projection->id()] = $projection;
+    }
+
+    public function remove(Projection $projection): void
+    {
+        unset($this->projections[$projection->id()]);
+    }
+
+    public function transactional(Closure $closure): void
+    {
+        $closure();
     }
```

Comment:
This is here not really transactional. We would first need to answer the question: Should this class be used outside of tests? And if yes, which could be done i think, then we should implement this to match the behaviour which the user expects it like.

Reply:
I solved the problem with "TransactionalStore" interface.